SU-HSDO/suhumsci
# STN-544: add file and pdf icon links to hb and legacy theme

# READY FOR REVIEW

## Summary
This card adds pdf and file link styling to add icons to links selected within the WYSIWYG/CKEditor when editing Drupal pages. These styles utilize the Font Awesome icons, so they should match those that would have been used with the Font Awesome library previously. This also gives the option to remove the icon from the link, by removing the link style if needed, as this was something that was difficult using the Font Awesome icon plugin on CKEditor previously.

Jira Card: [STN-544](https://sparkbox.atlassian.net/browse/STN-544)

## Need Review By (Date)
9/18

## Urgency
medium

## Steps to Test
1. Run npm test
2. Import changes to the CKEditor config files.
3. Create a basic page, (content > basic page) and add "PDF Link" and a "File Link" to the page within the WYSIWYG editor link style options. The text will need to be added as a link/anchor before the PDF and File link styles are available from the style dropdown.
![image](https://user-images.githubusercontent.com/9355698/93547884-1427f500-f934-11ea-8efd-bf3921a123ae.png)
4. Confirm the link icons/styles show up in the WYSIWYG while content editing and the option in text styles exist. These links and icons should update color with the theme color pairings (these links are using the secondary color to the pairings).
5. These icons should be able to be swapped/modified if you choose a different link style, additionally, make sure the icon goes away if you delete the link.
![image](https://user-images.githubusercontent.com/9355698/93548007-52251900-f934-11ea-910f-fba7d46457c5.png)
6. Confirm the styling shows up correctly on the page view after saving the page.
![image](https://user-images.githubusercontent.com/9355698/93548126-944e5a80-f934-11ea-8f2d-602702900571.png)
7. Confirm these options are all available with all editor entry options: (Basic HTML, Basic HTML w/o Media, Full HTML, & Minimal HTML). 
8. Additional testing options would be to add links to cards, wells, and within a spotlight.
9. Test that same page in the different appearance themes: Colorful, Traditional, and the Legacy theme.

Note: An "outline" version of the icon has been added as a class modifier but is not added to the link style dropdown, it has been added for possible future use upon request.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
